test/e2e/e2e_test.go

```diff
@@ -72,4 +81,48 @@ var _ = Describe("JobSet Operator", Ordered, func() {
 			return nil
 		}, 5*time.Minute, 5*time.Second).Should(Succeed(), "available condition is not found")
 	})
+
+	It("Verifying operand pod deleted and recovery", func() {
+		ctx := context.TODO()
+		pods, err := clients.KubeClient.CoreV1().Pods(operatorNamespace).List(ctx, metav1.ListOptions{
+			LabelSelector: operandLabel,
+		})
+		Expect(err).NotTo(HaveOccurred())
+		Expect(pods.Items).ToNot(BeEmpty())
+		err = clients.KubeClient.CoreV1().Pods(operatorNamespace).DeleteCollection(
+			ctx,
+			metav1.DeleteOptions{
+				GracePeriodSeconds: ptr.Int64(30),
+			},
+			metav1.ListOptions{
+				LabelSelector: operandLabel,
+			},
+		)
+		Expect(err).NotTo(HaveOccurred())
+
+		// Wait recovery(Deployment will recreate Pod)
+		err = wait.Poll(2*time.Second, 30*time.Second, func() (bool, error) {
+			newPods, err := clients.KubeClient.CoreV1().Pods(operatorNamespace).List(ctx, metav1.ListOptions{
+				LabelSelector: operandLabel,
+			})
+			if err != nil {
+				if errors.IsNotFound(err) {
```

Comment:
Can we actually get `IsNotFound` err when fetching for a pod list?

Reply:
Yes, you are right, should no  IsNotFound err  exists.